bpf/lib/nodeport.h

```diff
@@ -663,14 +663,16 @@ int tail_nodeport_nat_ingress_ipv6(struct __ctx_buff *ctx)
 
 	ret = snat_v6_rev_nat(ctx, &target);
 	if (IS_ERR(ret)) {
-		/* In case of no mapping, recircle back to main path. SNAT is very
-		 * expensive in terms of instructions (since we don't have BPF to
-		 * BPF calls as we use tail calls) and complexity, hence this is
-		 * done inside a tail call here.
-		 */
-		ctx_skip_nodeport_set(ctx);
-		ep_tail_call(ctx, CILIUM_CALL_IPV6_FROM_NETDEV);
-		ret = DROP_MISSED_TAIL_CALL;
+		if (ret == DROP_NAT_NOT_NEEDED || ret == DROP_NAT_NO_MAPPING) {
```

Comment:
How come this doesn't handle https://github.com/cilium/cilium/blob/07dea5d1da0337ff2e5c27ef993a1e8f5fa41047/bpf/lib/nat.h#L1513-L1517 ?

Reply:
For this case we want to drop the packet. I think we are good now. (In opposite to previously where the tail-call was hit for no reasons)

Reply:
> For this case we want to drop the packet. I think we are good now. (In opposite to previously where the tail-call was hit for no reasons)

Why is it ok to drop such packets? Where would they get dropped by the main path otherwise?